src/readpool/mod.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod task;
+
+use futures_cpupool::{Builder as CpuPoolBuilder, CpuPool};
+
+use storage::Engine;
+use server::Config;
+
+pub use self::task::{BoxedFuture, Error, Priority, Result, Task, Value};
+pub use self::task::kvget::*;
+pub use self::task::kvbatchget::*;
+pub use self::task::coprocessor::*;
+
+pub struct WorkerThreadContext {
+    engine: Box<Engine>,
+}
+
+impl Clone for WorkerThreadContext {
+    fn clone(&self) -> WorkerThreadContext {
+        WorkerThreadContext {
+            engine: self.engine.clone(),
+        }
+    }
+}
+
+pub struct ReadPool {
```

Comment:
Please use an individual pr for readpool.

Reply:
@zhangjinpeng1987 Do you mean the PR should exclude KvGet, KvBatchGet, etc?

Reply:
Exactly.